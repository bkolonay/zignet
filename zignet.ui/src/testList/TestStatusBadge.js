import React, { Component } from 'react';

class TestStatusBadge extends Component {

  render() {
    const badge = this.props.failingFromDate ? 
      <span className='badge badge-danger'>Down</span> :
      <span className='badge badge-success'>Up</span>;

    return (
      badge
    );
  }
}

export default TestStatusBadge;
