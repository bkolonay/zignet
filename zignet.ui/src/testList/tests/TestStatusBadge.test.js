import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import TestStatusBadge from '../TestStatusBadge';

it('renders when has failing from date', () => {
  const component = <TestStatusBadge failingFromDate={'2018-05-24T01:00:00'} />;
  ReactDOM.render(component,document.createElement('div'));
});

it('renders when does not have failing from date', () => {
  const component = <TestStatusBadge failingFromDate={undefined} />;
  ReactDOM.render(component,document.createElement('div'));
});

it('snapshot when has failing from date', () => {
  const component = <TestStatusBadge failingFromDate={'2018-05-24T01:00:00'} />;
  const tree = renderer.create(component).toJSON();
  expect(tree).toMatchSnapshot();
});

it('snapshot when does not have failing from date', () => {
  const component = <TestStatusBadge failingFromDate={undefined} />;
  const tree = renderer.create(component).toJSON();
  expect(tree).toMatchSnapshot();
});