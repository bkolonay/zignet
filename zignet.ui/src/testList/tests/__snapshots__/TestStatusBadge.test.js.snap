// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot when does not have failing from date 1`] = `
<span
  className="badge badge-success"
>
  Up
</span>
`;

exports[`snapshot when has failing from date 1`] = `
<span
  className="badge badge-danger"
>
  Down
</span>
`;
